src/main/scala/mesosphere/marathon/core/instance/Instance.scala

```diff
@@ -258,13 +257,12 @@ object Instance {
       goal: Goal): InstanceState = {
 
       val tasks = newTaskMap.values
-
       // compute the new instance condition
       val condition = conditionFromTasks(tasks, now, unreachableStrategy)
-
+      // TODO: set condition to Scheduled when task is terminal and goal = Running?
```

Comment:
What do you mean?

Reply:
I would drop that TODO because I don't like design options captured as comments but it means exactly that - when all tasks are terminal, instance condition is terminal. Then `isScheduled` picks that up...

So an instance in such terminal state is scheduled but it's condition is not `Scheduled` but e.g. `Killed`

For me the right answer here is that we should drop `Scheduled` as a condition altogether at some point and use just `isScheduled`

Reply:
Agreed :)

There's a lot of overlap with the information contained in a Mesos task status and the instance goal. IE, for task status running and instance goal running, the condition could be computed as running. For task status running and instance goal stopped, the condition could be computed as killing.

I'd like to see a future where condition is entirely derived from other state.

Reply:
if we get rid of the Scheduled as a condition, it'd be interesting to know the implications it would have on the event stream: IE do we publish an event when the instance becomes scheduled?

Reply:
I think we do publish that. Good point. We should note that once we create JIRA for removing `Scheduled` - if we decide to go that way. I'll leave this one unresolved to figure that out...

Reply:
I thought we never used the `Scheduled` condition since we define

```scala
val isScheduled: Boolean = state.goal == Goal.Running && !isActive && !state.condition.isLost && state.condition != UnreachableInactive
```

AFAIK we only use it in `Task.Update`.

Reply:
no, we use it. Look at `Instance.scheduled` method